diff --git a/meson.build b/meson.build
index 0744ba1..95c09c3 100644
--- a/meson.build
+++ b/meson.build
@@ -138,18 +138,12 @@ if host_machine.system() == 'windows' and not python3_dep.found()
 endif
 
 pygobject_dep = dependency('pygobject-3.0', version: pygobject_req, required: false)
-lua51_dep = dependency('lua51', version: lua_req, required: false)
-if not lua51_dep.found()
-  lua51_dep = dependency('lua-5.1', version: lua_req, required: false)
-endif
-luajit_dep = dependency('luajit', version: luajit_req, required: false)
+lua51_dep = dependency('lua', version: lua_req, required: get_option('lua51'))
+luajit_dep = dependency('luajit', version: luajit_req, required: get_option('luajit'))
 lua_lgi_found = false
 lua_lgi_ver = 'not found'
-lua51_prg = find_program('lua5.1', required: false)
-if not lua51_prg.found()
-  lua51_prg = find_program('lua51', required: false)
-endif
-luajit_prg = find_program('luajit', required: false)
+lua51_prg = find_program('lua', required: get_option('lua51'))
+luajit_prg = find_program('luajit', required: get_option('luajit'))
 xmllint_prg = find_program('xmllint', required: false)
 generate_vapi = get_option('vapi')
 vapigen_prg = find_program('vapigen', required: generate_vapi)
@@ -211,6 +205,8 @@ if lua_found
             lua51_lgi_dep = declare_dependency(version: lua_lgi_ver)
             lua_lgi_found = true
         endif
+    else
+        error('lua support requested but lua-lgi not found')
     endif
     message('lua-lgi version: ' + lua_lgi_ver)
 endif
@@ -394,11 +390,7 @@ endif
 
 build_gjs_loader = get_option('gjs')
 
-build_lua51_loader = get_option('lua51')
-lua51_found = (luajit_dep.found() or lua51_dep.found()) and lua_lgi_found
-if build_lua51_loader and not lua51_found
-  error('Lua51 requested but failed to locate suitable Lua51 and LGI support')
-endif
+build_lua51_loader = (luajit_dep.found() or lua51_dep.found()) and lua_lgi_found
 
 build_python3_loader = get_option('python3')
 python3_found = python3_dep.found() and pygobject_dep.found()
diff --git a/meson_options.txt b/meson_options.txt
index 48a7e8d..891bf2a 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -3,9 +3,13 @@ option('gjs',
        description: 'Enable GJS support (requires gjs-1.0)')
 
 option('lua51',
-       type: 'boolean', value: true,
+       type: 'feature', value: 'enabled',
        description: 'Enable Lua 5.1 support (requires lua-lgi)')
 
+option('luajit',
+       type: 'feature', value: 'disabled',
+       description: 'Use LuaJIT for Lua 5.1 support (requires lua-lgi)')
+
 option('python3',
        type: 'boolean', value: true,
        description: 'Enable Python 3 support (requires pygobject)')
